R/checks.R

```diff
@@ -789,4 +789,45 @@ check_fracwith_diswith <- function(form, rep) {
     filter(ptstatus == 1 & (num_of_fractures > 1 |
                               location_of_fracture != location_of_dislocation))
   return(problems)
+}
+
+#' Check that the time from injury to def. stabilization should be within
+#'  1 hour of the time from injury to abx administration
+#' 
+#' @param form dataframe containing ptstatus, form 4.1, and form5.3
+#' @return a dataframe containing problematic entries with relevant columns
+#' @import tidyverse
+#' @export 
+check_locof1stadmin_pritosur <- function(form) {
+  hrswithin <- 1
+  problems <- form %>% 
+    transmute(
+      region, site, studyid, ptinit, ptstatus, ihhrs, ishrs_1, iahrs,
+      itoshrs = ihhrs + ishrs_1, 
+      comment = "The time from injury to def. stabilization should be within 1 hour of the time from injury to abx administration") %>% 
+    filter(ptstatus == 1 & !(abs(itoshrs - iahrs) <= hrswithin | ihhrs < 0 | ishrs_1 < 0 | iahrs < 0))
+  return(problems)
+}
+
+#' Check that the time from injury to abx administration should be no more than
+#'  12 hours from the time from injury to def. stabilization
+#' 
+#' @param form dataframe containing ptstatus, form 4.1, and form5.3
+#' @return a dataframe containing problematic entries with relevant columns
+#' @import tidyverse
+#' @export 
+check_locof1stadmin_ope <- function(form) {
+  hrsmore <- 12
+  problems <- form %>% 
+    transmute(
+      region, site, studyid, ptinit, ptstatus, 
+      ihhrs = coalesce(pull(form, ihhrs), 0), ihdays = coalesce(pull(form, ihdays), 0),
+      ishrs_1 = coalesce(pull(form, ishrs_1), 0), isdays_1 = coalesce(pull(form, isdays_1), 0),
+      iahrs = coalesce(pull(form, iahrs), 0), iadays = coalesce(pull(form, iadays), 0),
+      hrsdiff = (iahrs + iadays * 24) - (ihhrs + ihdays * 24 + ishrs_1 + isdays_1 * 24),
+      comment = "The time from injury to abx administration should be no more than 12 hours from the time from injury to def. stabilization") %>% 
+    filter(ptstatus == 1 &
+             !(ihhrs <0 | ihdays <0 | ishrs_1 < 0 | isdays_1 < 0 | iahrs < 0 | iadays < 0 | 
```

Comment:
No need to check for 0 or other edge cases. They will be checked in other functions later.

Reply:
I did this because ihhrs, ihdays and the other two sets of hrs and days can take the values -1, -2 NA etc., which would mess up the calculation for hrsdiff. So I used coalesce to change all NA to 0, and the above filters to filter out rows with negative values. I couldn't figure out a better way to do this.